arcgis-ios-sdk-samples/Layers/Browse OGC API feature service/BrowseOGCAPIFeatureServiceViewController.swift

```diff
@@ -0,0 +1,246 @@
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class BrowseOGCAPIFeatureServiceViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemapStyle: .arcGISTopographic)
+            // Set the viewpoint to Daraa, Syria.
+            mapView.setViewpoint(AGSViewpoint(latitude: 32.62, longitude: 36.10, scale: 20_000))
+        }
+    }
+    /// The bar button to browse feature layers.
+    @IBOutlet var layersBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties
+    
+    /// The most recent query job.
+    var lastQuery: AGSCancelable?
+    /// The Daraa, Syria OGC API feature service URL.
+    let defaultServiceURL = URL(string: "https://demo.ldproxy.net/daraa")!
+    /// The service metadata of feature collections from the loaded service.
+    var featureCollectionInfos: [AGSOGCFeatureCollectionInfo] = []
+    /// The OGC feature collection info selected by the user.
+    var selectedInfo: AGSOGCFeatureCollectionInfo?
+    /// The OGC API features service.
+    var service: AGSOGCFeatureService!
+    /// A flag to indicate whether the service URL has been provided.
+    var urlProvided = false
+    
+    /// The query parameters to populate features from the OGC API service.
+    let queryParameters: AGSQueryParameters = {
+        let queryParameters = AGSQueryParameters()
+        // Set a limit of 1000 on the number of returned features per request,
+        // because the default on some services could be as low as 10.
+        queryParameters.maxFeatures = 1_000
+        return queryParameters
+    }()
+    
+    // MARK: Methods
+    
+    /// Create and load the OGC API features service from a URL.
+    func loadService(url: URL) {
+        service = AGSOGCFeatureService(url: url)
+        service.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else if let infos = self.service.serviceInfo?.featureCollectionInfos {
+                self.featureCollectionInfos = infos
+                self.layersBarButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    /// Load and display a feature layer from the OGC feature collection table.
+    /// - Parameter table: The `AGSOGCFeatureCollectionTable` selected by user.
+    func displayLayer(with table: AGSOGCFeatureCollectionTable) {
```

Comment:
Didn't this class take the info before? You can go back to that as there's really no difference between it taking the info or the table. I did do some testing and apparently _something_ (the service?) is holding onto the table, as it doesn't go away when the completion handler holds a weak reference to it. However, it would be good for the completion handler to hold an _unowned_ reference to the table.

Reply:
Changing it back to take `info` as parameter, IMO the local `table` variable needs to be held strongly. 

I tried adding `unowned` and it seems to crash with completion handler. But I cannot be too sure if there are other objects that hold the table. Please let me know your thoughts!